packages/@aws-cdk/cdk/lib/aspects/README.md

```diff
@@ -0,0 +1,292 @@
+## Aspects
+
+Aspects are a mechanism to extend the CDK without having to direclty impact the
+class heirarchy directly. This is a form of [Aspect-oriented
+programming](https://en.wikipedia.org/wiki/Aspect-oriented_programming) which we
+have implemented using the [Visitor
+Pattern](https://en.wikipedia.org/wiki/Aspect-oriented_programming).
+
+### Aspects in the CDK
+
+An aspect in the CDK is defined by the interface:
+
+```ts
+/**
+ * Represents an Aspect
+ */
+export interface IAspect {
+  /**
+   * Aspects invocation pattern
+   */
+  readonly visitType: AspectVisitType;
+  /**
+   * All aspects can visit a IConstruct
+   */
+  visit(node: IConstruct): void;
+}
+```
+
+Initially two types of aspects are supported. 
+
+1. Single visit aspects will ensure the aspect visit function is only invoked
+   one time.
+1. Multiple visit aspects will invoke the visit function as many times as
```

Comment:
I don't understand what this means though.

Reply:
I am ready to rip out the this concept. @rix0rrr @eladb -- looking for agreement from you guys.

Also this means the interface for aspect is simplified to:

```ts
export interface IAspect {
  /**
   * All aspects can visit a IConstruct
   */
  visit(node: IConstruct): void;
}
```

Are you good with that too?